test/microbench/org/apache/cassandra/test/microbench/instance/WriteTest.java

```diff
@@ -0,0 +1,302 @@
+                            "CREATE TABLE %s ( userid bigint, picid bigint, commentid bigint, PRIMARY KEY(userid, picid)) with compression = {'enabled': false}" +
+                            memtableSetup);
+        execute("use " + keyspace + ";");
+        switch (async)
+        {
+        case SERIAL_NET:
+        case PARALLEL_NET:
+            CQLTester.requireNetwork();
+            executeNet(getDefaultVersion(), "use " + keyspace + ";");
+        }
+        writeStatement = "INSERT INTO " + table + "(userid,picid,commentid)VALUES(?,?,?)";
+        System.err.println("Prepared, batch " + BATCH + " flush " + flush);
+        System.err.println("Disk access mode " + DatabaseDescriptor.getDiskAccessMode() + " index " +
+                           DatabaseDescriptor.getIndexAccessMode());
+
+        cfs = Keyspace.open(keyspace).getColumnFamilyStore(table);
+        cfs.disableAutoCompaction();
+        cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.USER_FORCED);
+    }
+
+    @Benchmark
+    public void writeTable() throws Throwable
+    {
+        long i;
+        for (i = 0; i <= count - BATCH; i += BATCH)
+            performWrite(i, BATCH);
+        if (i < count)
+            performWrite(i, count - i);
+
+        switch (flush)
+        {
+        case FLUSH:
+            cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.USER_FORCED);
+            // if we flush we also must truncate to avoid accummulating sstables
+        case TRUNCATE:
+            execute("TRUNCATE TABLE " + table);
+            // note: we turn snapshotting and durable writes (which would have caused a flush) off for this benchmark
+            break;
+        case INMEM:
+            if (!cfs.getLiveSSTables().isEmpty())
+                throw new AssertionError("SSTables created for INMEM test.");
+            // leave unflushed, i.e. next iteration will overwrite data
+        default:
+        }
+    }
+
+    public Object[] writeArguments(long i)
+    {
+        return new Object[] { i, i, i };
+    }
+//    abstract Object[] writeArguments(long i);
+
+    public void performWrite(long ofs, long count) throws Throwable
+    {
+        switch (async)
+        {
+        case SERIAL:
+            performWriteSerial(ofs, count);
+            break;
+        case SERIAL_NET:
+            performWriteSerialNet(ofs, count);
+            break;
+        case PARALLEL:
+            performWriteThreads(ofs, count);
+            break;
+        case PARALLEL_NET:
+            performWriteThreadsNet(ofs, count);
+            break;
+        }
+    }
+
+    public void performWriteSerial(long ofs, long count) throws Throwable
+    {
+        for (long i = ofs; i < ofs + count; ++i)
+            execute(writeStatement, writeArguments(i));
+    }
+
+    public void performWriteThreads(long ofs, long count) throws Throwable
+    {
+        long done = LongStream.range(ofs, ofs + count)
+                              .parallel()
```

Comment:
It is okay, in the sense that it is giving repeatable results with parallelism.

But since you asked and I already have one, updated with a version that lets one specify the number of threads to use.